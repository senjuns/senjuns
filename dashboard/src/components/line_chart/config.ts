import { Config, Layout, PlotData } from 'plotly.js';
import { XAxis, YAxis, LineState } from './types';

const PLOTLY_LINE_CHART_AXIS_AND_TABLE_COLOR = 'rgba(0,0,0,0.1)';

/**
 * Rgba colors for the Line chart shapes.
 */
export enum ChartColors {
  RED_LOW = 'rgba(221, 56, 47, 0.15)',
  RED_HIGH = 'rgba(221, 56, 47, 0.15)',
  YELLOW_LOW = 'rgba(255, 185, 76, 0.1)',
  YELLOW_HIGH = 'rgba(255, 185, 76, 0.1)',
  GREEN = 'rgba(49, 202, 113, 0.1)',
  GREEN_LINE = '#159199',
  RED_LINE = '#dd382f',
}

/**
 * Configuration for the Plotly Line chart.
 *
 * @returns {Partial<Config>} - The Plotly line configuration.
 */
export const plotlyConfig = (): Partial<Config> => {
  return {
    scrollZoom: true,
    responsive: true,
    displaylogo: false,
    modeBarButtonsToRemove: [
      'toggleSpikelines',
      'lasso2d',
      'select2d',
      'pan2d',
      'zoom2d',
      'zoomIn2d',
      'zoomOut2d',
      'hoverCompareCartesian',
      'hoverClosestCartesian',
      'autoScale2d',
    ],
  };
};

/**
 * Configuration of the Plotly chart layout.
 * @param {XAxis[]} xRange - Range which will be shown in chart for X axis.
 * @param {YAxis[]} yRange - Range which will be shown in chart for Y axis.
 * @param {any[]} shapes - Shapes that will be shown in chart, this is being used for horizontal and vertical shapes.
 * @returns {Partial<Layout>} - The Plotly layout configuration.
 */
export const plotlyLayoutConfig = (
  xRange: XAxis[],
  yRange: YAxis[],
  shapes: any[]
): Partial<Layout> => {
  return {
    yaxis: {
      fixedrange: true,
      range: yRange,
      tickcolor: PLOTLY_LINE_CHART_AXIS_AND_TABLE_COLOR,
      gridcolor: PLOTLY_LINE_CHART_AXIS_AND_TABLE_COLOR,
      zerolinecolor: PLOTLY_LINE_CHART_AXIS_AND_TABLE_COLOR,
    },
    xaxis: {
      constraintoward: 'right',
      type: 'date',
      range: xRange,
      tickcolor: PLOTLY_LINE_CHART_AXIS_AND_TABLE_COLOR,
      gridcolor: PLOTLY_LINE_CHART_AXIS_AND_TABLE_COLOR,
      tickangle: 0,
      showgrid: false,
    },
    dragmode: 'pan',
    hovermode: 'closest',
    margin: { t: 0, b: 50, r: 0, l: 50 },
    font: {
      family: 'Poppins',
      size: 13,
    },
    shapes,
    autosize: true,
  };
};

/**
 * Plotly configuration for the Line chart lines.
 * @param {LineState} state - State of the Line chart widget.
 * @returns {Partial<PlotData>[]} - The Plotly data configuration.
 */
export const plotlyDataConfig = (state: LineState): Partial<PlotData>[] => {
  return [
    {
      name: 'in range',
      x: state.xNorm,
      y: state.yNorm,
      type: 'scatter',
      hoverinfo: 'x+y',
      marker: { color: ChartColors.GREEN_LINE },
      connectgaps: false,
      showlegend: false,
    },
    {
      name: 'out of range',
      x: state.xAbnormal,
      y: state.yAbnormal,
      type: 'scatter',
      hoverinfo: 'x+y',
      marker: { color: ChartColors.RED_LINE },
      connectgaps: false,
      showlegend: false,
    },
  ];
};

/**
 * Returns configuration for the Plotly line chart horizontal shape.
 * @param {number} yAxisStartPoint - Start of shape on y axis.
 * @param {number} yAxisEndPoint - End of shape on y axis.
 * @param {string} backgroundColor - Color for the the shape and its body.
 * @param {Date | number} xAxisStartPoint - Start of shape on x axis.
 * @param {Date | number} xAxisEndPoint - End of shape on x axis.
 * @returns {any} - Configuration of the shape.
 */
export const getHorizontalShapeConfig = (
  yAxisStartPoint: number,
  yAxisEndPoint: number,
  backgroundColor: string,
  xAxisStartPoint: Date | number = 0,
  xAxisEndPoint: Date | number = 1
): any => {
  return {
    type: 'rect',
    layer: 'below',
    fillcolor: backgroundColor,
    line: {
      width: 0,
    },
    xsizemode: 'scaled',
    ysizemode: 'scaled',
    xref: 'x',
    x0: xAxisStartPoint,
    x1: xAxisEndPoint,
    yref: 'y',
    y0: yAxisStartPoint,
    y1: yAxisEndPoint,
  };
};

/**
 * Returns configuration for the Plotly line chart vertical shape.
 * @param {Date | number} xAxisStartPoint - Start of shape on x axis.
 * @param {Date | number} xAxisEndPoint - End of shape on y axis.
 * @param {string} backgroundColor - Color for the the shape and its body.
 * @param {Date | number} yAxisStartPoint - Start of shape on y axis.
 * @param {Date | number} yAxisEndPoint - End of shape on y axis.
 * @returns {any} - Configuration of the shape.
 */
export const getVerticalShapeConfig = (
  xAxisStartPoint: Date | number,
  xAxisEndPoint: Date | number,
  backgroundColor: string,
  yAxisStartPoint: number = 0,
  yAxisEndPoint: number = 1
): any => {
  return {
    type: 'rect',
    layer: 'below',
    fillcolor: backgroundColor,
    line: {
      width: 0,
    },
    xsizemode: 'scaled',
    ysizemode: 'scaled',
    xref: 'x',
    x0: xAxisStartPoint,
    x1: xAxisEndPoint,
    yref: 'paper',
    y0: yAxisStartPoint,
    y1: yAxisEndPoint,
  };
};
