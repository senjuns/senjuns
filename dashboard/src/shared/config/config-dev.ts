export const config: any = {
  gaTrackingId: 'AB-1234568-9',
  aws_hasura_graphqlEndpoint:
    'https://dashboard-hasura.dev.neatleaf.com/v1/graphql',
  aws_project_region: 'us-west-2',
  aws_cognito_identity_pool_id:
    'COGNITO_DASHBOARD_APP_USER_POOL_IDENTITY_POOL_ID',
  aws_cognito_region: 'us-west-2',
  aws_user_pools_id: 'USER_POOL_ID',
  aws_user_pools_web_client_id: 'USER_POOL_WEB_CLIENT',
  aws_appsync_graphqlEndpoint:
    'https://uijj43arhnc2hmjcycwjzt3eyu.appsync-api.us-west-2.amazonaws.com/graphql',
  aws_appsync_region: 'us-west-2',
  aws_appsync_authenticationType: 'AMAZON_COGNITO_USER_POOLS',
  aws_appsync_apiKey: 'da2-fakeApiId123456',
  aws_content_delivery_bucket: 'ladybeetle-hostingbucket-dev',
  aws_content_delivery_bucket_region: 'us-west-2',
  aws_content_delivery_url: 'https://d2ewqb7falxfxe.cloudfront.net',
  Auth: {
    identityPoolId: 'COGNITO_DASHBOARD_APP_USER_POOL_IDENTITY_POOL_ID',
    region: 'us-west-2',
    userPoolId: 'USER_POOL_ID',
    userPoolWebClientId: 'USER_POOL_WEB_CLIENT',
  },
  Storage: {
    AWSS3: {
      bucket: 'dev-spyder-v1-imagebucket',
      region: 'us-west-2',
    },
  },
  oauth: {
    domain: 'neatleaf-enmo-test.auth.us-west-2.amazoncognito.com',
    scope: ['email', 'openid'],
    redirectSignIn: 'http://localhost:3000/',
    redirectSignOut: 'http://localhost:3000/',
    responseType: 'code',
  },
  federationTarget: 'COGNITO_USER_AND_IDENTITY_POOLS',
};

export default config;
