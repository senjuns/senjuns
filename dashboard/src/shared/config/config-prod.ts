const config = {
  gaTrackingId: 'AB-1234568-9',
  aws_project_region: 'eu-central-1',
  aws_cognito_identity_pool_id:
    'COGNITO_DASHBOARD_APP_USER_POOL_IDENTITY_POOL_ID',
  aws_cognito_region: 'eu-central-1',
  aws_user_pools_id: 'USER_POOL_ID',
  aws_user_pools_web_client_id: 'USER_POOL_WEB_CLIENT',
  aws_appsync_graphqlEndpoint: 'AWS_APPSYNC_GRAPHQLENDPOINT',
  aws_appsync_region: 'eu-central-1',
  // aws_appsync_authenticationType: 'AMAZON_COGNITO_USER_POOLS',
  aws_appsync_authenticationType: 'AWS_IAM',
  Auth: {
    identityPoolId: 'COGNITO_DASHBOARD_APP_USER_POOL_IDENTITY_POOL_ID',
    region: 'eu-central-1',
    userPoolId: 'USER_POOL_ID',
    userPoolWebClientId: 'USER_POOL_WEB_CLIENT',
  },
};

export default config;
