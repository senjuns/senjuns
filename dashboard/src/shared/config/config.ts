export const config: any = {
  gaTrackingId: 'UA-179417847-2',
  aws_hasura_graphqlEndpoint:
    'https://dashboard-hasura.dev.neatleaf.com/v1/graphql',
  aws_project_region: 'us-west-2',
  aws_cognito_identity_pool_id:
    'us-west-2:6554116a-9656-4483-ba01-dc6d51ceed19',
  aws_cognito_region: 'us-west-2',
  aws_user_pools_id: 'us-west-2_3zgoEdJ3Z',
  aws_user_pools_web_client_id: '75s1peah34k3rmm307otr6ifpf',
  aws_appsync_graphqlEndpoint:
    'https://uijj43arhnc2hmjcycwjzt3eyu.appsync-api.us-west-2.amazonaws.com/graphql',
  aws_appsync_region: 'us-west-2',
  aws_appsync_authenticationType: 'AMAZON_COGNITO_USER_POOLS',
  aws_appsync_apiKey: 'da2-fakeApiId123456',
  aws_content_delivery_bucket: 'ladybeetle-hostingbucket-dev',
  aws_content_delivery_bucket_region: 'us-west-2',
  aws_content_delivery_url: 'https://d2ewqb7falxfxe.cloudfront.net',
  Auth: {
    identityPoolId: 'us-west-2:6554116a-9656-4483-ba01-dc6d51ceed19',
    region: 'us-west-2',
    userPoolId: 'us-west-2_3zgoEdJ3Z',
    userPoolWebClientId: '75s1peah34k3rmm307otr6ifpf',
  },
  Storage: {
    AWSS3: {
      bucket: 'dev-spyder-v1-imagebucket',
      region: 'us-west-2',
    },
  },
  oauth: {
    domain: 'neatleaf-enmo-test.auth.us-west-2.amazoncognito.com',
    scope: ['email', 'openid'],
    redirectSignIn: 'http://localhost:3000/',
    redirectSignOut: 'http://localhost:3000/',
    responseType: 'code',
  },
  federationTarget: 'COGNITO_USER_AND_IDENTITY_POOLS',
};

export default config;
