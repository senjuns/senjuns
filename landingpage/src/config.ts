export type TConfig = {
  MAIL_SUBMIT_API: string;
};

const configDev: TConfig = {
  MAIL_SUBMIT_API: 'https://mail.dev.neatleaf.com',
};

const configProd: TConfig = {
  MAIL_SUBMIT_API: 'https://mail.neatleaf.com',
};

let config: TConfig = configDev;
if (
  process.env.REACT_APP_STAGE === 'prod' ||
  process.env.REACT_APP_STAGE === 'production'
) {
  config = configProd;
}

export default config;
