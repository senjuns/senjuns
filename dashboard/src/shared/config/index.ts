import Cookies from 'js-cookie';
// import { FEATURE_ENABLED } from '../../shared/constants';
import { FEATURE_FLAGS } from '../../shared/interfaces';
// @ts-ignore Needs to be ignored as it is dynamically created.
import config from './config';

const defaultConfig: any = config;

if (Cookies.get(FEATURE_FLAGS.DEBUG) === 'FEATURE_ENABLED') {
  console.log(`process.env: ${JSON.stringify(process.env, null, 2)}`);
  console.log(`defaultConfig: ${JSON.stringify(defaultConfig, null, 2)}`);
}

export default defaultConfig;
