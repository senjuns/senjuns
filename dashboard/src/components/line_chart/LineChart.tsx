import MomentUtils from '@date-io/moment';
import CircularProgress from '@material-ui/core/CircularProgress';
import { DatePicker, MuiPickersUtilsProvider } from '@material-ui/pickers';
import React, { lazy, useEffect, useState } from 'react';
import styled from 'styled-components';
import { ReactComponent as CalendarIcon } from '../../assets/svg/calendar.svg';
import { ReactComponent as DropdownArrowIcon } from '../../assets/svg/dropdown-arrow.svg';
import { Typography } from '../../components/common/Typography';
import { useCropCycleDetails } from '../../contexts';
import { useFeatureFlags } from '../../contexts/FeatureFlagsProvider';
import { useGetLatestMeasurementRunBySystemIdQuery } from '../../graphql/generated/react_apollo';
import { Colors } from '../../shared/constants';
import { MeasurementTypes } from '../../shared/interfaces';
import { getCurrentDateTime } from '../../shared/utils';
import {
  ChartColors,
  getHorizontalShapeConfig,
  getVerticalShapeConfig,
} from './config';
import { useLineChartData } from './useLineChartData';
import { ILineChartPlotParams } from './useLineChartPlotInfo';
import { getMeasurementSpecificData, splitIntoRedAndGreenLines } from './utils';

const LineChartView = lazy(() => import('./LineChartView'));
const MeasurementTypeBar = lazy(
  () => import('../../components/measurement_type_bar/MeasurementTypeBar')
);
const NoDataView = lazy(() => import('../../components/error/NoDataView'));

const momentUtils = new MomentUtils();

interface LineChartProps {
  systemId: number;
}

const LineChart: React.FC<LineChartProps> = ({ systemId }) => {
  const { airflowSensorFlag, calendarFlag } = useFeatureFlags();
  const { range } = useCropCycleDetails();
  const [measurementOption, setMeasurementOption] = useState(
    MeasurementTypes.Temperature
  );

  const { data: latestMeasurementRun } =
    useGetLatestMeasurementRunBySystemIdQuery({
      variables: {
        system_id: { _eq: systemId },
        start: range.start,
        end: range.end,
      },
    });

  useEffect(() => {
    if (!latestMeasurementRun || !latestMeasurementRun.measurement_run?.[0])
      return;

    const latestMeasurementRunDate = new Date(
      latestMeasurementRun.measurement_run[0].start_time
    );
    const delta = 3;
    const startDate = new Date(latestMeasurementRunDate);
    startDate.setDate(startDate.getDate() - delta);

    setSelectedDate(momentUtils.date(latestMeasurementRunDate).format('lll'));
  }, [latestMeasurementRun]);

  const { measurementRunData, growthCycleData, loading, error } =
    useLineChartData({
      zoneId: 1,
      systemId,
      start: range.start,
      end: range.end,
    });

  const hasNoData = !loading && !measurementRunData?.length;

  const currentFormattedDateString = momentUtils
    .date(getCurrentDateTime())
    .format('lll');
  const [selectedDate, setSelectedDate] = useState<string | null>(
    currentFormattedDateString
  );

  const [isDatePickerVisible, setDatePickerVisible] = useState(false);

  const [lineChartViewParams, setLineChartViewParams] =
    useState<ILineChartPlotParams>({
      xRange: [],
      yRange: [],
      shapes: [],
      state: {
        data: { timestamps: [], values: [] },
        xNorm: [],
        yNorm: [],
        xAbnormal: [],
        yAbnormal: [],
      },
    });

  useEffect(() => {
    if (!measurementRunData || !growthCycleData) return;

    const data = getMeasurementSpecificData(
      measurementRunData,
      growthCycleData,
      measurementOption
    );

    if (!data) return;

    const { xNorm, yNorm, xAbnormal, yAbnormal } = splitIntoRedAndGreenLines(
      data.data,
      (data.monitoringParams?.light.yellow as any)?.min,
      (data.monitoringParams?.light.yellow as any)?.max
    );

    const shapes = [];
    // add day & night cycle area

    for (
      let date = new Date(range.start);
      date <= range.end;
      date.setDate(date.getDate() + 1)
    ) {
      // previous night
      const prevNightEndTime = new Date(date);
      prevNightEndTime.setHours(6);
      prevNightEndTime.setMinutes(0);
      prevNightEndTime.setSeconds(0);
      const prevNightStartTime = new Date(prevNightEndTime);
      prevNightStartTime.setDate(prevNightStartTime.getDate() - 1);
      prevNightStartTime.setHours(18);
      prevNightStartTime.setMinutes(0);
      prevNightStartTime.setSeconds(0);

      shapes.push(
        getVerticalShapeConfig(prevNightStartTime, prevNightEndTime, Colors.hb6)
      );

      // next night
      const nextNightStartTime = new Date(prevNightStartTime);
      nextNightStartTime.setDate(prevNightStartTime.getDate() + 1);
      const nextNightEndTime = new Date(prevNightEndTime);
      nextNightEndTime.setDate(prevNightEndTime.getDate() + 1);
      shapes.push(
        getVerticalShapeConfig(nextNightStartTime, nextNightEndTime, Colors.hb6)
      );

      // dark cycle
      // add green rectangle area
      shapes.push(
        getHorizontalShapeConfig(
          (data.monitoringParams?.dark.green as any)?.min,
          (data.monitoringParams?.dark.green as any)?.max,
          ChartColors.GREEN,
          prevNightStartTime,
          prevNightEndTime
        )
      );
      if (date === range.end) {
        shapes.push(
          getHorizontalShapeConfig(
            (data.monitoringParams?.dark.green as any)?.min,
            (data.monitoringParams?.dark.green as any)?.max,
            ChartColors.GREEN,
            nextNightStartTime,
            nextNightEndTime
          )
        );
      }

      // add yellow rectangle area
      shapes.push(
        getHorizontalShapeConfig(
          (data.monitoringParams?.dark.yellow as any)?.min,
          (data.monitoringParams?.dark.green as any)?.min,
          ChartColors.YELLOW_LOW,
          prevNightStartTime,
          prevNightEndTime
        )
      );
      shapes.push(
        getHorizontalShapeConfig(
          (data.monitoringParams?.dark.yellow as any)?.max,
          (data.monitoringParams?.dark.green as any)?.max,
          ChartColors.YELLOW_LOW,
          prevNightStartTime,
          prevNightEndTime
        )
      );
      if (date === range.end) {
        shapes.push(
          getHorizontalShapeConfig(
            (data.monitoringParams?.dark.yellow as any)?.min,
            (data.monitoringParams?.dark.green as any)?.min,
            ChartColors.YELLOW_LOW,
            nextNightStartTime,
            nextNightEndTime
          )
        );
        shapes.push(
          getHorizontalShapeConfig(
            (data.monitoringParams?.dark.green as any)?.max,
            (data.monitoringParams?.dark.yellow as any)?.max,
            ChartColors.YELLOW_HIGH,
            nextNightStartTime,
            nextNightEndTime
          )
        );
      }

      // light cycle
      // add green rectangle area
      shapes.push(
        getHorizontalShapeConfig(
          (data.monitoringParams?.light.green as any)?.min,
          (data.monitoringParams?.light.green as any)?.max,
          ChartColors.GREEN,
          prevNightEndTime,
          nextNightStartTime
        )
      );

      // add yellow rectangle area
      shapes.push(
        getHorizontalShapeConfig(
          (data.monitoringParams?.light.yellow as any)?.min,
          (data.monitoringParams?.light.green as any)?.min,
          ChartColors.YELLOW_LOW,
          prevNightEndTime,
          nextNightStartTime
        )
      );
      shapes.push(
        getHorizontalShapeConfig(
          (data.monitoringParams?.light.yellow as any)?.max,
          (data.monitoringParams?.light.green as any)?.max,
          ChartColors.YELLOW_LOW,
          prevNightEndTime,
          nextNightStartTime
        )
      );
    }

    setLineChartViewParams({
      xRange: [range.start, range.end],
      yRange: [data.min.toString(), data.max.toString()],
      shapes,
      state: { data: data.data, xNorm, yNorm, xAbnormal, yAbnormal },
    });
  }, [measurementOption, loading]);

  const handleDateChange = (date: any) => {
    setSelectedDate(date.format('lll'));
    setDatePickerVisible(false);
  };

  const handleCalendarClick = () => {
    setDatePickerVisible(!isDatePickerVisible);
  };

  if (error) {
    return <Typography variant="h6">{JSON.stringify(error)}</Typography>;
  }

  if (loading) {
    return <CircularProgress />;
  }

  if (hasNoData) {
    return <NoDataView />;
  }

  return (
    <Container>
      <MeasurementTypeBar
        airflowSensorFlag={airflowSensorFlag}
        value={measurementOption}
        onOptionChange={setMeasurementOption}
      />
      <Typography variant="h6">{measurementOption}</Typography>
      <DateRow>
        <MeasurementDate>{selectedDate}</MeasurementDate>
        {calendarFlag && (
          <CalendarDropdown>
            <CalendarButton onClick={handleCalendarClick}>
              <CalendarIcon />
              <DropdownArrowIcon />
            </CalendarButton>

            <DatePickerContainer visible={isDatePickerVisible}>
              <MuiPickersUtilsProvider utils={MomentUtils}>
                <DatePicker
                  disableToolbar
                  variant="static"
                  value={selectedDate}
                  onChange={handleDateChange}
                />
              </MuiPickersUtilsProvider>
            </DatePickerContainer>
          </CalendarDropdown>
        )}
      </DateRow>
      <LineChartView
        xRange={lineChartViewParams.xRange}
        yRange={lineChartViewParams.yRange}
        shapes={lineChartViewParams.shapes}
        state={lineChartViewParams.state}
      />
      <Legend>
        <InRange>
          <GreenCircle />
          <LegendTitle>In range</LegendTitle>
        </InRange>
        <OutOfRange>
          <RedCircle />
          <LegendTitle>Out of range</LegendTitle>
        </OutOfRange>
      </Legend>
    </Container>
  );
};

const Container = styled.div`
  display: flex;
  flex-direction: column;
  gap: 12px;
`;

const MeasurementDate = styled.p`
  font-style: normal;
  font-weight: normal;
  font-size: 16px;
`;

const Legend = styled.div`
  display: flex;
  justify-content: flex-end;
  gap: 20px;
`;

const InRange = styled.div`
  display: flex;
  gap: 5px;
  align-items: center;
`;

const OutOfRange = styled.div`
  display: flex;
  gap: 5px;
  align-items: center;
`;

const GreenCircle = styled.div`
  background: ${ChartColors.GREEN_LINE};
  width: 6px;
  height: 6px;
  border-radius: 3px;
`;

const RedCircle = styled.div`
  background: ${ChartColors.RED_LINE};
  width: 6px;
  height: 6px;
  border-radius: 3px;
`;

const LegendTitle = styled.p`
  font-size: 12px;
`;

const CalendarDropdown = styled.div`
  position: relative;
`;

const CalendarButton = styled.div`
  cursor: pointer;
  display: flex;
  gap: 15px;
  justify-content: center;
  align-items: center;
  width: fit-content;
  padding: 16px;
  background-color: ${Colors.hb6};
  border-radius: 20px;
`;

const DateRow = styled.div`
  display: flex;
  justify-content: space-between;
  margin-bottom: 30px;
`;

const DatePickerContainer = styled.div<{ visible: boolean }>`
  position: absolute;
  width: fit-content;
  height: fit-content;
  display: ${(props) => (props.visible ? 'block' : 'none')};
  right: 0;
  z-index: 10;
  border: 1px solid black;
`;

export default LineChart;
