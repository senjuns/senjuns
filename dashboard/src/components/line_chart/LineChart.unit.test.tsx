import {
  ApolloClient,
  ApolloProvider,
  createHttpLink,
  InMemoryCache,
} from '@apollo/client';
import { setContext } from '@apollo/client/link/context';
// eslint-disable-next-line import/no-extraneous-dependencies
import * as ShallowRenderer from 'react-test-renderer/shallow';
import config from '../../shared/config';
import LineChart from './LineChart';

const httpLink = createHttpLink({
  uri: config.aws_hasura_graphqlEndpoint,
});

const authLink = setContext((_, { headers }) => {
  return {
    headers: {
      ...headers,
      'x-hasura-admin-secret':
        process.env.REACT_APP_HASURA_GRAPHQL_ADMIN_SECRET,
    },
  };
});
const client = new ApolloClient({
  link: authLink.concat(httpLink),
  cache: new InMemoryCache(),
});

describe('<LineChart />', () => {
  it('renders correctly.', async () => {
    const renderer = ShallowRenderer.createRenderer();
    renderer.render(
      <ApolloProvider client={client}>
        <LineChart systemId={2} />
      </ApolloProvider>
    );

    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});
