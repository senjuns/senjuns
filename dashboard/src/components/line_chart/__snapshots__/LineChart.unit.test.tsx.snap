// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LineChart /> renders correctly. 1`] = `
<Context.Consumer>
  [Function]
</Context.Consumer>
`;
